codex-rs/execpolicy2/src/main.rs

```diff
@@ -0,0 +1,54 @@
+use std::fs;
+use std::path::Path;
+
+use anyhow::Context;
+use anyhow::Result;
+use clap::Parser;
+use codex_execpolicy2::PolicyParser;
+
+/// CLI for evaluating exec policies
+#[derive(Parser)]
+#[command(name = "codex-execpolicy2")]
+enum Cli {
+    /// Evaluate a command against a policy.
+    Check {
+        #[arg(short, long, value_name = "PATH")]
+        policy: String,
+
+        /// Command tokens to check.
+        #[arg(
+            value_name = "COMMAND",
+            required = true,
+            trailing_var_arg = true,
+            allow_hyphen_values = true
+        )]
+        command: Vec<String>,
+    },
+}
+
+fn main() -> Result<()> {
+    let cli = Cli::parse();
+    match cli {
+        Cli::Check { policy, command } => cmd_check(policy, command),
+    }
+}
+
+fn cmd_check(policy_path: String, args: Vec<String>) -> Result<()> {
+    let policy = load_policy(&policy_path)?;
+
+    let eval = policy.check(&args);
+    let json = serde_json::to_string_pretty(&eval)?;
+    println!("{json}");
+    Ok(())
+}
+
+fn load_policy(policy_path: &str) -> Result<codex_execpolicy2::Policy> {
+    let content = fs::read_to_string(policy_path).with_context(|| {
+        format!(
+            "failed to read policy at {}",
+            Path::new(policy_path).display()
+        )
+    })?;
+    let parser = PolicyParser::new(policy_path, &content);
+    Ok(parser.parse()?)
```

Comment:
Can you just do this?

```suggestion
    parser.parse()
```